hopsworks-api/src/main/java/io/hops/hopsworks/api/featurestore/FeaturestoreService.java

```diff
@@ -1228,5 +1248,53 @@ public Response updateTrainingDatasetMetadata(@Context SecurityContext sc,
       .entity(trainingDatasetGeneric)
       .build();
   }
+  
+  /**
+   * Verify that the user is allowed to execute the requested operation based on its project role
+   *
+   * Only data owners are allowed to update/delete feature groups created by someone else in the project
+   *
+   * @param featuregroupDTO the featuregroup that the operation concerns
+   * @param featurestore the featurestore that the operation concerns
+   * @param user the user requesting the operation
+   * @throws FeaturestoreException
+   */
+  private void verifyUserRoleForFeaturegroupOperation(FeaturegroupDTO featuregroupDTO, Featurestore featurestore,
+    Users user)
+    throws FeaturestoreException {
+    String userRole = projectTeamFacade.findCurrentRole(project, user);
+    if(!featuregroupDTO.getCreator().equals(user.getEmail()) &&  userRole !=
+      AllowedRoles.DATA_OWNER){
+      throw new FeaturestoreException(RESTCodes.FeaturestoreErrorCode.UNAUTHORIZED_FEATURESTORE_OPERATION, Level.SEVERE,
```

Comment:
@Limmen That doesn't look like a `SEVERE` log message, more like `FINE` as it's an unauthorized operation due to a wrong client request.

Reply:
Done